AQ0X_TESTER_CONFIG/doxygen/man/man3/Console.3
.TH "Console" 3 "Thu Oct 30 2014" "Version V0.0" "AQ0X" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Console \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <console\&.h>\fP
.PP
Inherits \fBQPlainTextEdit\fP\&.
.SS "Signals"

.in +1c
.ti -1c
.RI "void \fBgetData\fP (const QByteArray &data)"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBConsole\fP (QWidget *parent=0)"
.br
.ti -1c
.RI "void \fBputData\fP (const QByteArray &data)"
.br
.ti -1c
.RI "void \fBsetLocalEchoEnabled\fP (bool set)"
.br
.ti -1c
.RI "QString \fBuint8_to_str\fP (quint8 val)"
.br
.in -1c
.SS "Public Attributes"

.in +1c
.ti -1c
.RI "QPalette \fBp\fP"
.br
.in -1c
.SS "Protected Member Functions"

.in +1c
.ti -1c
.RI "virtual void \fBkeyPressEvent\fP (QKeyEvent *e)"
.br
.ti -1c
.RI "virtual void \fBmousePressEvent\fP (QMouseEvent *e)"
.br
.ti -1c
.RI "virtual void \fBmouseDoubleClickEvent\fP (QMouseEvent *e)"
.br
.ti -1c
.RI "virtual void \fBcontextMenuEvent\fP (QContextMenuEvent *e)"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 48 of file console\&.h\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "Console::Console (QWidget *parent = \fC0\fP)\fC [explicit]\fP"

.PP
Definition at line 49 of file console\&.cpp\&.
.PP
References p\&.
.PP
.nf
50     : QPlainTextEdit(parent)
51     , localEchoEnabled(false)
52 {
53     document()->setMaximumBlockCount(100);
54     p = palette();
55     p\&.setColor(QPalette::Base, Qt::black);
56     p\&.setColor(QPalette::Text, Qt::green);
57     setPalette(p);
58 
59 }
.fi
.SH "Member Function Documentation"
.PP 
.SS "void Console::contextMenuEvent (QContextMenuEvent *e)\fC [protected]\fP, \fC [virtual]\fP"

.PP
Definition at line 110 of file console\&.cpp\&.
.PP
.nf
111 {
112     Q_UNUSED(e)
113 }
.fi
.SS "void Console::getData (const QByteArray &data)\fC [signal]\fP"

.PP
Definition at line 120 of file moc_console\&.cpp\&.
.PP
Referenced by keyPressEvent()\&.
.PP
.nf
121 {
122     void *_a[] = { 0, const_cast<void*>(reinterpret_cast<const void*>(&_t1)) };
123     QMetaObject::activate(this, &staticMetaObject, 0, _a);
124 }
.fi
.SS "void Console::keyPressEvent (QKeyEvent *e)\fC [protected]\fP, \fC [virtual]\fP"

.PP
Definition at line 82 of file console\&.cpp\&.
.PP
References getData()\&.
.PP
.nf
83 {
84     switch (e->key()) {
85     case Qt::Key_Backspace:
86     case Qt::Key_Left:
87     case Qt::Key_Right:
88     case Qt::Key_Up:
89     case Qt::Key_Down:
90         // skip processing
91         break;
92     default:
93         if (localEchoEnabled)
94             QPlainTextEdit::keyPressEvent(e);
95         emit getData(e->text()\&.toLocal8Bit());
96     }
97 }
.fi
.SS "void Console::mouseDoubleClickEvent (QMouseEvent *e)\fC [protected]\fP, \fC [virtual]\fP"

.PP
Definition at line 105 of file console\&.cpp\&.
.PP
.nf
106 {
107     Q_UNUSED(e)
108 }
.fi
.SS "void Console::mousePressEvent (QMouseEvent *e)\fC [protected]\fP, \fC [virtual]\fP"

.PP
Definition at line 99 of file console\&.cpp\&.
.PP
.nf
100 {
101     Q_UNUSED(e)
102     setFocus();
103 }
.fi
.SS "void Console::putData (const QByteArray &data)"

.PP
Definition at line 61 of file console\&.cpp\&.
.PP
References uint8_to_str()\&.
.PP
.nf
62 {
63 
64     QString str;
65 
66 
67     for(int ii =0;ii<data\&.length();ii++)
68         str\&.append(uint8_to_str((quint8)data\&.at(ii)));
69 
70 
71     insertPlainText(QString(data));
72 
73     QScrollBar *bar = verticalScrollBar();
74     bar->setValue(bar->maximum());
75 }
.fi
.SS "void Console::setLocalEchoEnabled (boolset)"

.PP
Definition at line 77 of file console\&.cpp\&.
.PP
.nf
78 {
79     localEchoEnabled = set;
80 }
.fi
.SS "QString Console::uint8_to_str (quint8val)"

.PP
Definition at line 116 of file console\&.cpp\&.
.PP
Referenced by putData()\&.
.PP
.nf
117 {
118     const char cnv[] = "0123456789ABCDEF";
119 
120     return QString(cnv[(val>>4)&0x0F]) + QString(cnv[(val>>0)&0x0F]);
121 }
.fi
.SH "Member Data Documentation"
.PP 
.SS "QPalette Console::p"

.PP
Definition at line 64 of file console\&.h\&.
.PP
Referenced by Console()\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for AQ0X from the source code\&.


AQ0X_TESTER_CONFIG/project/AQ0X/console.cpp
/****************************************************************************
**
** Copyright (C) 2012 Denis Shienkov <denis.shienkov@gmail.com>
** Copyright (C) 2012 Laszlo Papp <lpapp@kde.org>
** Contact: http://www.qt-project.org/legal
**
** This file is part of the QtSerialPort module of the Qt Toolkit.
**
** $QT_BEGIN_LICENSE:LGPL$
** Commercial License Usage
** Licensees holding valid commercial Qt licenses may use this file in
** accordance with the commercial license agreement provided with the
** Software or, alternatively, in accordance with the terms contained in
** a written agreement between you and Digia.  For licensing terms and
** conditions see http://qt.digia.com/licensing.  For further information
** use the contact form at http://qt.digia.com/contact-us.
**
** GNU Lesser General Public License Usage
** Alternatively, this file may be used under the terms of the GNU Lesser
** General Public License version 2.1 as published by the Free Software
** Foundation and appearing in the file LICENSE.LGPL included in the
** packaging of this file.  Please review the following information to
** ensure the GNU Lesser General Public License version 2.1 requirements
** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
**
** In addition, as a special exception, Digia gives you certain additional
** rights.  These rights are described in the Digia Qt LGPL Exception
** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
**
** GNU General Public License Usage
** Alternatively, this file may be used under the terms of the GNU
** General Public License version 3.0 as published by the Free Software
** Foundation and appearing in the file LICENSE.GPL included in the
** packaging of this file.  Please review the following information to
** ensure the GNU General Public License version 3.0 requirements will be
** met: http://www.gnu.org/copyleft/gpl.html.
**
**
** $QT_END_LICENSE$
**
****************************************************************************/

#include "console.h"

#include <QScrollBar>

#include <QtCore/QDebug>

Console::Console(QWidget *parent)
    : QPlainTextEdit(parent)
    , localEchoEnabled(false)
{
    document()->setMaximumBlockCount(100);
    p = palette();
    p.setColor(QPalette::Base, Qt::black);
    p.setColor(QPalette::Text, Qt::green);
    setPalette(p);

}

void Console::putData(const QByteArray &data)
{

    QString str;


    for(int ii =0;ii<data.length();ii++)
        str.append(uint8_to_str((quint8)data.at(ii)));


    insertPlainText(QString(data));

    QScrollBar *bar = verticalScrollBar();
    bar->setValue(bar->maximum());
}

void Console::setLocalEchoEnabled(bool set)
{
    localEchoEnabled = set;
}

void Console::keyPressEvent(QKeyEvent *e)
{
    switch (e->key()) {
    case Qt::Key_Backspace:
    case Qt::Key_Left:
    case Qt::Key_Right:
    case Qt::Key_Up:
    case Qt::Key_Down:
        // skip processing
        break;
    default:
        if (localEchoEnabled)
            QPlainTextEdit::keyPressEvent(e);
        emit getData(e->text().toLocal8Bit());
    }
}

void Console::mousePressEvent(QMouseEvent *e)
{
    Q_UNUSED(e)
    setFocus();
}

void Console::mouseDoubleClickEvent(QMouseEvent *e)
{
    Q_UNUSED(e)
}

void Console::contextMenuEvent(QContextMenuEvent *e)
{
    Q_UNUSED(e)
}


QString Console::uint8_to_str(quint8 val)
{
    const char cnv[] = "0123456789ABCDEF";

    return QString(cnv[(val>>4)&0x0F]) + QString(cnv[(val>>0)&0x0F]);
}
